MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -194,29 +204,33 @@ public void UpdateEncoderPath(string path, string pathType)
         /// <returns><c>true</c> if the version validation succeeded; otherwise, <c>false</c>.</returns>
         private bool ValidatePath(string path, FFmpegLocation location)
         {
-            bool rc = false;
-
             if (!string.IsNullOrEmpty(path))
             {
                 if (File.Exists(path))
                 {
-                    rc = new EncoderValidator(_logger, path).ValidateVersion();
-
-                    if (!rc)
+                    var version = _encoderValidator.ValidateVersion(path);
+                    if (version != null)
                     {
-                        _logger.LogWarning("FFmpeg: {Location}: Failed version check: {Path}", location, path);
+                        if (EncoderValidator.BestVersion(version, _bestVersion))
```

Comment:
Couple of things:

1. `BestVersion` is not a great function name and neither is the return value. It's very hard to decipher what it does.
2. The xmldocs are incorrect or the usage is incorrect: `<returns><c>false</c> if source is better, <c>true</c> if destination is better.</returns>`.
3. `ValidateVersion` returning a Version instead of a bool is also not great.

I would suggest that `BestVersion` is changed to something like `TryUpdateVersion` that conditionally updates the "current" version in the validator, then after the "loop" is completed, the path etc. are pulled from the validator into the the media encoder.

Reply:
changed names. Hopefully, it will be easier to follow now.